### PR TITLE
Add mute commands

### LINKED ISSUE #40: Adding moderation commands: mute, unmute, tempmute
In this issue, I'd like the mute, unmute, and tempmute commands to be added to the bot

If you're unfamiliar with the commands, you can test them out first hand, but this is a basic run down of what should happen.

#### Command format
```
cc!mute @ user reason
cc!unmute @ user
cc!tempmute @ user lengthoftime reason
```

#### When the bot joins
A new role called "Muted" should be created with no permissions to speak anywhere.

#### When the command is run
The bot should add the role to the user. When the mute command is called, it should just leave the role on the user until the unmute command is called.
When the tempmute command is called, the user should be given the muted role for a specified length of time. The format of this length of time should be along the format of seconds to years, using the first letter of the word as a label e.g. 60s = 60 seconds, 3w = 3 weeks, etc, etc. At the end of that time, the role should be removed from the user.

#### Limited Use
Please limit the use of the mute command to Moderators and Admins.
Please limit the use of the unmute and tempmute commands to Super Users, Moderators, and Admins.

Make sure to check on #7 before getting started so you don't miss anything

### PR DESCRIPTION
**Pull request by Victoria-DR**
# Description
Add **cc!mute**, **cc!unmute**, and **cc!tempmute** commands. These commands mute, unmute, and temporarily mute the user by making use of a _Muted_ role, log the use of these commands in **infractions** and **mod_log**, and sends DMs to users notifying them of the muted/unmuted status. Closes #40.

# Commands
## cc!mute
Use the following command, where `user` is a mention (i.e. @user) and `reason` is text.
```
cc!mute [user] [reason]
```
This command adds a **Muted** role to the mentioned user and denies them message sending and role reacting privileges. A confirmation message is sent to the same channel the command was used in and a DM is sent to the user notifying them of their muted status. A log of each use of this command is added to both the **infractions** and **mod_log** tables.

## cc!unmute
Use the following command, where `user` is a mention (i.e. @user).
```
cc!mute [user]
```
This command removes a **Muted** role from the mentioned user and grants them message sending and role reacting privileges. A confirmation message is sent to the same channel the command was used in and a DM is sent to the user notifying them of their unmuted status. A log of each use of this command is added to both the **infractions** and **mod_log** tables.

## cc!tempmute
Use the following command, where `user` is a mention (i.e. @user), `lengthoftime` is a period of time, and `reason` is text. `lengthoftime` should use the format `[number][unit]` where `number` is any number and `unit` is a single lowercase letter (either s for seconds, m for minutes, h for hours, d for days, w for weeks, or y for years).
```
cc!mute [user] [lengthoftime] [reason]
```
This command adds a **Muted** role from the mentioned user and denies them message sending and role reacting privileges. A confirmation message is sent to the same channel the command was used in and a DM is sent to the user notifying them of their muted status. A log of this command is added to both the **infractions** and **mod_log** tables. After the specified time period, this command removes the **Muted** role from the user and grants them message sending and role reacting privileges. A confirmation message is sent to the same channel the command was used in and a DM is sent to the user notifying them of their muted status. A log of the unmuting of the user is added to both the **infractions** and **mod_log** tables with the reason provided as "tempmute expired" and moderator as "automatic".

### CODE REVIEW

**Comment by aedwardg** [comment]
@Victoria-DR, 

Can you provide some insight on how we can test this and what we should be looking for when testing out the added functionality?

Thanks!

**Comment by Vic-ST** [comment]
@aedwardg I would check the issue #40, it has a pretty good list of what commands are supposed to work.

**Comment by Victoria-DR** [comment]
@aedwardg I just updated the original message for the pull request to include this info.

File: commands/mute/mute.js
Comment (Vic-ST) [inline commands/mute/mute.js:79]: Why don't you put this all on one line separated by a semi-colon? You'll have to include `multipleStatements: true` on the query.

File: commands/mute/tempmute.js
Comment (Vic-ST) [inline commands/mute/tempmute.js:91]: Same here, might as well put it all on one line instead of two different queries

File: commands/mute/tempmute.js
Comment (Vic-ST) [inline commands/mute/tempmute.js:100]: Won't setting the timestamp outside of the setTimeout function set it to the time that the command was run instead of the time he was unmuted?

image

File: commands/mute/tempmute.js
Comment (Vic-ST) [inline commands/mute/tempmute.js:125]: Same thing... might want to put this all on one line

File: commands/mute/unmute.js
Comment (Vic-ST) [inline commands/mute/unmute.js:68]: One last one

File: commands/mute/tempmute.js
Comment (Vic-ST) [inline commands/mute/tempmute.js:21]: Could you outsource the regex to a variable name, just to allow for better readability?

**Review by Vic-ST — CHANGES_REQUESTED** [review]
:tada:

Works like a charm! Just a couple things I'd like fixed...

File: commands/mute/mute.js
Comment (Victoria-DR) [reply commands/mute/mute.js]: Done

File: commands/mute/tempmute.js
Comment (Victoria-DR) [reply commands/mute/tempmute.js]: Done

File: commands/mute/tempmute.js
Comment (Victoria-DR) [reply commands/mute/tempmute.js]: Done

File: commands/mute/unmute.js
Comment (Victoria-DR) [reply commands/mute/unmute.js]: Done

File: commands/mute/tempmute.js
Comment (Victoria-DR) [reply commands/mute/tempmute.js]: Done

**Review by Vic-ST — APPROVED** [review]
LGTM! We'll have to remember to add this to the help command....

**Comment by Victoria-DR** [comment]
@Vic-ST Yep, I think the ban and kick commands are done/nearly done, so we can add them all at once?

**Comment by Vic-ST** [comment]
@Victoria-DR Yes, I'm going to wait until you merge yours into master+dev so I can rebase before creating a PR

File: commands/mute/mute.js
Comment (aedwardg) [inline commands/mute/mute.js:60]: In the future, I think it would be better to have `NULL` rather than `'N/A'` for missing values.
No need to change this now though.

File: commands/mute/tempmute.js
Comment (aedwardg) [inline commands/mute/tempmute.js:21]: `command_regex` should probably be moved above this conditional so it can be changed to:
```javascript
if (!args.join(" ").match(command_regex)
```

File: commands/mute/unmute.js
Comment (aedwardg) [inline commands/mute/unmute.js:22]: Are we doing anything with `toUnmuteDisplay`? Do we need it? If not, let's get rid of it.

**Review by aedwardg — CHANGES_REQUESTED** [review]
Great job @Victoria-DR! This looks like it does what it's supposed to do once everything is set up properly.

I left a few comments--some are tiny changes, some are questions and some are just a head's up for moving forward with the project.

One final thing I want to bring up is that we should probably be breaking these up into smaller chunks (this can be its own issue, no need to do it right now).  Each of these commands can be broken up into 3-4 aptly-named functions that will make it easier to understand and update in the future.

For example, the mute command's `execute` function could look more like this:
```javascript
execute(msg, con) {
    let toMute = undefined;
    let reason = undefined;

    if (!msg.member.roles.cache.some(
        role => role.name === "Moderator" || role.name === "Admin")) {
            return msg.reply("You must be a moderator or admin to use this command.");
    } else {
        const [status, muteMessage] = canMute();  // returns [bool, string]
        if (!status) {
            return msg.reply(muteMessage);
        }
        muteMember(muteMessage);
        sendEmbed();
        updateDB();
    }
}
```

File: app.js
Comment (Vic-ST) [reply app.js:33]: @aedwardg Hmmmm, so should we leave it to two different queries for now?

File: app.js
Comment (Victoria-DR) [reply app.js]: Done

File: app.js
Comment (Victoria-DR) [reply app.js]: Done

File: commands/mute/mute.js
Comment (Victoria-DR) [reply commands/mute/mute.js]: Done

File: commands/mute/tempmute.js
Comment (Victoria-DR) [reply commands/mute/tempmute.js]: Done

File: commands/mute/unmute.js
Comment (Victoria-DR) [reply commands/mute/unmute.js]: Done

File: app.js
Comment (Victoria-DR) [reply app.js:33]: Refactored the code to be more modular but haven't changed the connection issue yet since it's still under discussion.

**Review by aedwardg — APPROVED** [review]
This works like a charm and is much closer to self-documenting.

Thank you for all the hard work and patience @Victoria-DR - this rocket is well-earned! 🚀

**Comment by Vic-ST** [comment]
So excited to see this merge, congratulations @Victoria-DR :rocket:

**Pull request merged**
